libyul/AsmParser.cpp

```diff
@@ -224,30 +303,25 @@ std::optional<std::pair<std::string_view, SourceLocation>> Parser::parseSrcComme
 		return {{tail, SourceLocation{}}};
 	}
 
-	std::optional<int> const sourceIndex = toInt(match[1].str());
-	std::optional<int> const start = toInt(match[2].str());
-	std::optional<int> const end = toInt(match[3].str());
-
-	if (!sourceIndex.has_value() || !start.has_value() || !end.has_value())
+	if (indexOutOfRange || (sourceIndex < 0 && sourceIndex != -1) || (start < 0 && start != -1) || (end < 0 && end != -1))
```

Comment:
```suggestion
	if (indexOutOfRange || sourceIndex < -1 || start < -1 || end < -1)
```
This is equivalent, no?